metacatalog/ext/base.py

```diff
@@ -75,3 +78,38 @@ class MetacatalogExtensionInterface(abc.ABC):
     @abc.abstractclassmethod
     def init_extension(cls):
         pass
+
+    @classmethod
+    def init_cli(cls, subparsers: '_SubParsersAction[ArgumentParser]', defaults: 'ArgumentParser') -> None:
+        """
+        Add a new :class:`ArgumentParser <argparse.ArgumentParser>` to the metacatalog CLI.
+        The main CLI argument parser will call the ``init_cli`` class method of all active
+        extensions and pass the main subparser to the init function. The second argument
+        is the :class:`ArgumentParser <argparse.ArgumentParser>`, which holds the default
+        arguments, that should affect all CLI functions.
+
+        Example
+        -------
+
+        .. code-block:: Python
+
+            from metacatalog.ext import MetacatalogExtensionInterface
+
+            class MyExt(MetacatalogExtensionInterface):
+                @classmethod
+                def init_extension(cls):
+                    pass
+                
+                @classmethod
+                def cli(cls, args):
+                    if not args.quiet:
+                        print(args.foo.upper())
+                    
+                @classmethod
+                def init_cli(subparsers, defaults):
```

Comment:
As you stated in #271, this line should be `def init_cli(cls, subparsers, defaults)`